### PR TITLE
feat: Add Rust binding for SimSIMD library

### PR DESCRIPTION
**Pull request by pplanel**
This pull request is the beginning of the Rust binding implementation. I would like guidance on reexporting the library's functionalities. 

This draft was created based on the header present in the `golang` binding.

If you have any specific questions or need guidance on a particular aspect of the implementation, feel free to provide more details, and I'll do my best to assist you!

## Description
This pull request adds Rust bindings for the simsimd C++ library.
The Rust code is generated using the `bindgen` crate, and the build process includes compiling the C++ source file into a static library. 
The Rust bindings provide utility functions for cosine, inner product, and squared Euclidean distance based on the Golang Cgo header file.

## Changes Made
- Added Rust bindings for simsimd library.
- Implemented self-contained build script (`build.rs`) to generate Rust bindings and compile C++ source.
- Created header file (`include/reexport.h`) for simsimd library.
- Provided utility functions for cosine, inner product, and squared Euclidean distance in `lib.rs`.
- Updated `.gitignore` to exclude generated Rust build artifacts.

## Checklist
- [x] Unit tests and/or integration tests are included.

### CODE REVIEW

**Comment by ashvardanian** [comment]
Hi, @pplanel! Thanks a lot for your effort!

There are a couple of things that catch the eye:

1. Let's not use bindgen and other high-level tools - see [this StringZilla PR](https://github.com/ashvardanian/StringZilla/pull/74]
2. Using traits is a better way to pick the right overload, adding a type-specific suffix is a force decision in Go, as it doesn't support overloading function names - see [USearch binding](https://github.com/unum-cloud/usearch/blob/main/rust/lib.rs)

Please let me know if you'd have time to patch those 🤗

**Comment by pplanel** [comment]
Hey, @ashvardanian. Thanks for your feedback!
I have addressed the points you raised, added docs and benchmark.

**Comment by pplanel** [comment]
Oh, and I updated the crate folder structure based on StringZilla and USearch crates.

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+fn main() {
+    // Docs.rs only target is x86_64
+    // `include/simsimd/types.h:119:9: error: _Float16 is not supported on this target`
```
Comment (ashvardanian) [inline build.rs:3]: We can define a `SIMSIMD_NATIVE_F16=0` macro, that will avoid using this type and will make compilation pass with most compilers.

File: build.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    if std::env::var("DOCS_RS").is_ok() {
+        return;
+    }
+
+    cc::Build::new()
+        .file("rust/lib.c")
+        .include("include")
+        .flag("-O3")
+        .warnings(false)
+        .compiler("/usr/bin/clang")
```
Comment (ashvardanian) [inline build.rs:13]: It doesn't seem like a good idea to hard-code the compiler.

**Review by ashvardanian — CHANGES_REQUESTED** [review]
1. It is also important to add a `prerelease.yml` task to run tests and a `release.yml` task to publish the crate, before we merge into `main`. Those can be found in USearch. Can you please add those?
2. It's great to have benchmark scripts and tests. Can you please add commands for running those in README.md?

File: build.rs (hunk 1 above)
Comment (pplanel) [reply build.rs:3]: Nice, but hard-coded in the `build.rs` or include conditionally?

**Review by ashvardanian — APPROVED** [review]

**Pull request merged**